packages/deployments/contracts/contracts_forge/RelayerFeeRouter.t.sol

```diff
@@ -0,0 +1,205 @@
+      address(relayerFeeRouterImplementation),
+      abi.encodeWithSelector(RelayerFeeRouter.initialize.selector, xAppConnectionManager, connext)
+    );
+
+    relayerFeeRouter = RelayerFeeRouter(payable(address(proxy)));
+
+    // Shouldn't enrollRemoteRouter be protected by onlyOwner?
+    relayerFeeRouter.enrollRemoteRouter(remoteDomain, bytes32(remote));
+  }
+
+  // ============ initialize ============
+  function testInitializeParameters() public {
+    assertEq(address(relayerFeeRouter.xAppConnectionManager()), address(xAppConnectionManager));
+    assertEq(address(relayerFeeRouter.connext()), address(connext));
+  }
+
+  // ============ setConnext ============
+
+  // Should work
+  function testSetConnext() public {
+    vm.expectEmit(true, true, true, true);
+    emit SetConnext(connext2);
+
+    relayerFeeRouter.setConnext(connext2);
+    assertEq(address(relayerFeeRouter.connext()), connext2);
+  }
+
+  // Fail if not called by owner
+  function testSetConnextOnlyOwner() public {
+    vm.prank(address(0));
+    vm.expectRevert("Ownable: caller is not the owner");
+    relayerFeeRouter.setConnext(connext2);
+  }
+
+  // ============ send ============
+  // Fail if not called by connext
+  function testSendOnlyConnext() public {
+    vm.prank(address(0));
+    vm.expectRevert(abi.encodeWithSelector(RelayerFeeRouter.RelayerFeeRouter__onlyConnext_notConnext.selector));
+
+    bytes32[] memory transactionIds = new bytes32[](2);
+    transactionIds[0] = "AAA";
+    transactionIds[1] = "BBB";
+
+    relayerFeeRouter.send(remoteDomain, recipient, 1, transactionIds);
+  }
+
+  // Fail if not no amount and no transaction ids
+  function testSendClaimEmpty() public {
+    vm.expectRevert(abi.encodeWithSelector(RelayerFeeRouter.RelayerFeeRouter__send_claimEmpty.selector));
+
+    bytes32[] memory transactionIds = new bytes32[](0);
+
+    relayerFeeRouter.send(remoteDomain, recipient, 0, transactionIds);
+  }
+
+  // Fail if empty recipient
+  function testSendRecipientEmpty() public {
+    vm.expectRevert(abi.encodeWithSelector(RelayerFeeRouter.RelayerFeeRouter__send_recipientEmpty.selector));
+
+    bytes32[] memory transactionIds = new bytes32[](2);
+    transactionIds[0] = "AAA";
+    transactionIds[1] = "BBB";
+
+    relayerFeeRouter.send(remoteDomain, address(0), 1, transactionIds);
+  }
+
+  // Should work
+  function testSend(bytes32[] calldata _transactionIds, uint256 _amount) public {
+    vm.assume(_amount != 0 || _transactionIds.length != 0);
+    bytes memory message = RelayerFeeMessage.formatClaimFees(recipient, _amount, _transactionIds);
+
+    vm.expectCall(home, abi.encodeWithSelector(MockHome.dispatch.selector, remoteDomain, remote, message));
+    vm.expectEmit(true, true, true, true);
+    emit Send(remoteDomain, recipient, _amount, _transactionIds, remote, message);
+
+    relayerFeeRouter.send(remoteDomain, recipient, _amount, _transactionIds);
+  }
+
+  function testSend0Amount(bytes32[] calldata _transactionIds) public {
```

Comment:
Should make sure to add comments before every test thats more descriptive than the name.

I've been wondering if we should move to a test name convention of something like: `test_${ContractName}__${function}${test number for contract}` (i.e. `test_RelayerFeeRouter__send1`) so you can ensure when you run `test:forge -m ${name}` the test name is unique.

Thoughts?

Reply:
yeah, i agree, we should have any special naming rule for forge tests. it was very annoying to find some meaningful name for test function.

Reply:
we could do the descriptive name, that works for me!